### PR TITLE
[TASK] create ADR for decoupled transform library

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

ADR for decoupling the transform library from go code and allowing it to be used in TypeScript with Pepr. 

## Related Issue #1862 

Fixes #

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: adr/0017-decouple-transformation-library-from-zarf.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# NUMBER. TITLE
```
Comment (Racer159) [inline adr/0017-decouple-transformation-library-from-zarf.md:1]:
```suggestion
# NUMBER. TITLE
```
Fill this in.

File: adr/0017-decouple-transformation-library-from-zarf.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+Date: 2023-05-28
+
+## Status
+
+TBD
```
Comment (Racer159) [inline adr/0017-decouple-transformation-library-from-zarf.md:7]: Pending until your gRPC Pr is merged

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: adr/0017-decouple-transformation-library-from-zarf.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Context
+
+We need a way to decouple transformations from Zarf. We know the Zarf transformation library is battle tested, hardened, and reliable. As Pepr takes admission responsibilities from the Zarf Agent, we need a way to synchronize the TypeScript with the Go code instead of maintaining two disparate libraries which will be expected to grow.
```
Comment (Racer159) [inline adr/0017-decouple-transformation-library-from-zarf.md:11]: I think the context here should describe the three options a bit more:

1. WASM
2. gRPC
3. REST
4. Rewrite the code in Typescript

with simple pros and cons.

(then decision and consequences can be updated in the implementation PR)

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: adr/0017-decouple-transformation-library-from-zarf.md (hunk 2 above)
Comment (cmwylie19) [reply adr/0017-decouple-transformation-library-from-zarf.md:7]: Addressed in https://github.com/defenseunicorns/zarf/pull/1900

File: adr/0017-decouple-transformation-library-from-zarf.md (hunk 3 above)
Comment (cmwylie19) [reply adr/0017-decouple-transformation-library-from-zarf.md:11]: addressed in https://github.com/defenseunicorns/zarf/pull/1900

**Comment by cmwylie19** [comment]
Closed because there were major updates that needed to be made in the ADR so I recreated it since the content in this was stale.

**Pull request closed without merging**